### DISCUSSION TITLE
How do we scale tokens during inference time for silence prolongation

### DISCUSSION THREAD

**Discussion by JRMeyer** [coqui-ai/TTS#237] [Archived]
<USER>>>> georroussos<br/> <DATE> [July 16, 2020, 2:51pm]<br/> <br/> I am trying training using GST and it works very well. I find it makes<br/> the speech much more expressive and it can learn a few different styles;<br/> however, I would like to make the pauses in between sentences longer and<br/> even though my dataset has the exact same duration of pause mid periods,<br/> I haven't still figured out how to do it, as mentioned here<br/> <https://github.com/mozilla/TTS/issues/167>. My idea was that I could<br/> try multiplying the tensors by a value, but nothing really worked.<br/> Ideally, the style would stay the same, but the speech and pause would<br/> change. My point is that I do not really know how to disentangle these.<br/> <br/> [This is an archived TTS discussion thread from discourse.mozilla.org/t/how-do-we-scale-tokens-during-inference-time-for-silence-prolongation]<br/>

**Comment by JRMeyer** [comment]
<USER>>>> sanjaesc<br/> <DATE> [July 17, 2020, 9:47am]<br/> <br/> If you want to condition on tokens, you will need to query your gst<br/> layer for the specific token, which you can do as follows.<br/> <br/> init the query: gst_embedding_dim = embedding dim of your gst layer<br/> (like 256 or 512)<br/> <br/> > query = torch.zeros(1, 1, self.gst_embedding_dim//2).to(device)<br/> <br/> get your style tokens:<br/> <br/> > *slash* _GST = torch.tanh(self.gst_layer.style_token_layer.style_tokens)<br/> <br/> get the key for a specific token where TOKEN is some int value<br/> <br/> > key = *slash* _GST *slash* [TOKEN *slash* ].unsqueeze(0).expand(1, -1, -1)<br/> <br/> now you can query your gst attention layer for the specified key:<br/> <br/> > gst_outputs_att = self.gst_layer.style_token_layer.attention(query,<br/> > key)<br/> <br/> this gst_outputs_att *slash* [tensor *slash* ] can now be modified by multiplying:<br/> <br/> > gst_outputs = gst_outputs_att *slash* curl-run-all.sh discourse.mozilla.org html-to-markdown.sh ordered-posts ordered-posts~ TTS.cdx tts.commands tts-emails.txt TTS.pages tts-telegram.txt TTS.warc.gz 0.15 [ *slash* #example]<br/> <br/> from here you compute it just as if you were to condition on a wav file.<br/> <br/> Hope this helps.<br/> <br/> Check this commit<br/> <https://github.com/mozilla/TTS/pull/451/commits/c5eaf12d7b9b5b74b99a633a5622a44b123e49e4#><br/> for an example.<br/> <br/> [Archived Post]<br/>
